File: test-suite/python/security/jwt_verification_buggy.py
import jwt
from jose import jwt as jose_jwt
from jwt import decode as decode_jwt


def decode_without_signature(token: str):
    return jwt.decode(token, options={"verify_signature": False})


def decode_without_expiration(token: str, key: str):
    return jwt.decode(token, key, algorithms=["HS256"], options={"verify_exp": False})


def decode_without_verify(token: str, key: str):
    return jose_jwt.decode(token, key, algorithms=["HS256"], verify=False)


def decode_none_algorithm(token: str):
    return decode_jwt(token, None, ["none"])

File: test-suite/js/security/jwt-verification-buggy.ts
import jwt from "jsonwebtoken";
import { decode as decodeToken, verify } from "jsonwebtoken";
import { decodeJwt } from "jose";
import jwtDecode from "jwt-decode";

type Request = {
  headers: {
    authorization?: string;
  };
};

const SECRET = process.env.JWT_TEST_SECRET ?? "local-test-secret";

export function trustsDecodedRole(req: Request): string | undefined {
  const token = req.headers.authorization?.replace("Bearer ", "") ?? "";
  const claims = jwt.decode(token) as { role?: string } | null;
  return claims?.role;
}

export function trustsAliasedDecode(token: string): unknown {
  return decodeToken(token);
}

export function trustsJoseDecode(token: string): unknown {
  return decodeJwt(token);
}

export function trustsJwtDecodePackage(token: string): unknown {
  return jwtDecode(token);
}

export function verifiesButIgnoresExpiration(token: string): unknown {
  return verify(token, SECRET, { algorithms: ["HS256"], ignoreExpiration: true });
}

export function acceptsNoneAlgorithm(token: string): unknown {
  return jwt.verify(token, SECRET, { algorithms: ["none"] });
}
